cpp/src/strings/capitalize.cu

```diff
@@ -81,6 +149,25 @@ struct base_fn {
     auto const code_point = info.first;
     auto const flag       = info.second;
 
+    // first, check for the special title-case characters
+    auto tc_itr = thrust::upper_bound(
+                    thrust::seq, title_case_table.begin(), title_case_table.end(), code_point) -
+                  1;
+    if (*tc_itr == code_point) {
```

Comment:
Is it conceivable that `thrust::upper_bound()` returns `title_case_table.begin()`?  Then wouldn't the value of `tc_itr` become `title_case_table.begin() - 1`?

Wouldn't that make reading `*tc_itr` invalid?

Reply:
No, `thrust::upper_bound()` is not expected to return `.begin()`
https://nvidia.github.io/cccl/thrust/api/function_group__binary__search_1ga183fbd2f3d478f3e3d0dc97c5bce883b.html

Reply:
Sorry, I don't follow.  If the search key is less than the first/lowest value in `title_case_table`, wouldn't `upper_bound` return `.begin()`?

Reply:
Oh, I think you are right! I had moved code around and missed that now being possible.
Seems compute-sanitizer should've flagged this for me.

Reply:
Ah.  I was just about to post: https://godbolt.org/z/Mfo9asn9e